common/persistence/dataInterfaces.go

```diff
@@ -249,6 +251,18 @@ type (
 		ExpirationTime     time.Time
 		MaximumAttempts    int32
 		NonRetriableErrors []string
+		// events V2 related
+		EventStoreVersion   int32
+		CurrentResetVersion int32
+		HistoryTreeID       string
+		HistoryBranches     map[int32]*HistoryBranch // map from each resetVersion to the associated branch, resetVersion increase from 0
```

Comment:
Is this cached in-memory or written to the database?

Reply:
HistoryBranches will be written into database as part of mutableState. 
For now this map will only have one item.
Actually it is not required to have a map (resetVersion->branch) if we don't implement reset. We can just use a field of branch. But I am going to implement reset very soon, it is not good to have a field that will be deprecated.